fileseeker.go

```diff
@@ -42,16 +48,147 @@ type statikFile struct {
 	Time    time.Time `json:"time"`
 }
 
+type Config struct {
+	UpdateFrequency int    `toml:"update_frequency"`
+	Port            int    `toml:"port"`
+	DataPath        string `toml:"data_path"`
+	BaseUrl         string `toml:"root_url"`
+}
+
+type LockedFs struct {
+	fs webdav.FileSystem
+	mu *sync.RWMutex
+}
+
+func (lfs *LockedFs) OpenFile(ctx context.Context, name string, flag int, perm os.FileMode) (webdav.File, error) {
+	lfs.mu.RLock()
+
+	file, err := lfs.fs.OpenFile(ctx, name, flag, perm)
+	if err != nil {
+		lfs.mu.RUnlock()
+		return nil, err
+	}
+	return &LockedFile{file: file, mu: lfs.mu}, nil
+}
+
+func (lfs *LockedFs) Mkdir(ctx context.Context, name string, perm os.FileMode) error {
+	return errors.New("read-only filesystem: Mkdir is not allowed")
+}
+
+func (lfs *LockedFs) Rename(ctx context.Context, oldName, newName string) error {
+	return errors.New("read-only filesystem: Rename is not allowed")
+}
+
+func (lfs *LockedFs) RemoveAll(ctx context.Context, name string) error {
+	return errors.New("read-only filesystem: RemoveAll is not allowed")
+}
+
+func (lfs *LockedFs) Stat(ctx context.Context, name string) (os.FileInfo, error) {
+	return lfs.fs.Stat(ctx, name)
+}
+
+// needed to release RWMutex ()
+type LockedFile struct {
+	file webdav.File
+	mu   *sync.RWMutex
+}
+
+func (lf *LockedFile) Close() error {
+	err := lf.file.Close()
+	lf.mu.RUnlock()
+	return err
+}
+
+func (lf *LockedFile) Read(p []byte) (n int, err error) { return lf.file.Read(p) }
+
+func (lf *LockedFile) Write(p []byte) (n int, err error) {
+	return 0, errors.New("read-only filesystem: Write is not allowed")
+}
+
+func (lf *LockedFile) Seek(offset int64, whence int) (int64, error) {
+	return lf.file.Seek(offset, whence)
+}
+
+func (lf *LockedFile) Stat() (fs.FileInfo, error) { return lf.file.Stat() }
+
+func (lf *LockedFile) Readdir(count int) ([]fs.FileInfo, error) { return lf.file.Readdir(count) }
+
 var (
 	dataDir    = flag.String("d", "data", "data directory")
 	configFile = flag.String("c", "config/teachings.json", "config file")
 )
```

Comment:
Queste due variabili non servono più. La directory `data` è nel config file e la stringa `config/teachings.json` che sarebbe la lista degli insegnamenti andrebbe messa anche lei nel config file (magari chiamala `teachings_path`).

Ovviamente modifica il codice in cui queste variabili sono presenti in modo da usare quelle nuove